### PR TITLE
feat(api): Convert alert rules data model to be org level (SEN-972)

### PR DESCRIPTION
**Pull request by wedamija**
This just changes the data model to be able to handle things at the org level. I'll follow this up
with the logic changes to use the new fields, and remove the old ones.

### CODE REVIEW

File: src/sentry/incidents/models.py [hunk 1]
```diff
@@ -282,13 +303,11 @@ class AlertRule(Model):
...
     # QuerySubscription
     subscription_id = models.UUIDField(db_index=True, null=True)
-    dataset = models.TextField(null=True)
-    query = models.TextField(null=True)
-    aggregations = ArrayField(of=models.IntegerField, null=True)
-    time_window = models.IntegerField(null=True)
-    resolution = models.IntegerField(null=True)
+    aggregations = ArrayField(of=models.IntegerField)
+    query_subscription = FlexibleForeignKey("sentry.QuerySubscription", unique=True, null=True)
+    project = FlexibleForeignKey("sentry.Project", db_index=False, db_constraint=False, null=True)
```
Comment (markstory) [inline src/sentry/incidents/models.py]: If an organization alert rule is applied to multiple projects will we have multiple records in the `sentry_alertrule` table with similar data?

File: src/sentry/incidents/models.py [hunk 2]
```diff
@@ -263,16 +263,37 @@ def fetch_for_project(self, project):
...
-    query_subscription = FlexibleForeignKey("sentry.QuerySubscription", unique=True, null=True)
+    organization = FlexibleForeignKey("sentry.Organization", db_index=False, null=True)
+    query_subscriptions = models.ManyToManyField(
+        "sentry.QuerySubscription",
+        related_name="query_subscriptions",
+        through=AlertRuleQuerySubscription,
+    )
     name = models.TextField()
     status = models.SmallIntegerField(default=AlertRuleStatus.PENDING.value)
+    dataset = models.TextField()
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/incidents/models.py]: Does this map to a snuba dataset?

[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/incidents/models.py]: Yep, at first this will just be events

File: src/sentry/incidents/models.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/incidents/models.py]: No, there will just be the one org level row, with multiple related subscriptions for each project.

Comment (markstory) [reply src/sentry/incidents/models.py]: Are you converting this field to nullable to prepare it for future removal?

Comment (wedamija) [reply src/sentry/incidents/models.py]: Yeah, there's no need to have this field in the future since we'll have many projects related to each row.

**Review by markstory — APPROVED** [review]

**Pull request merged**